src/main/java/info/wzieleziecki/Graph.java

```diff
@@ -26,17 +29,18 @@ public Graph(List<String> wordList) {
         return getPath(previous, startWord, endWord);
     }
 
-    private List<String> getNextWords(String word) {
-        List<String> nextWords = new ArrayList<String>();
-        for (String currentWord : wordList) {
-            if (getWordDiffrenceNumber(word, currentWord) == 1) {
-                nextWords.add(currentWord);
+    private List<String> getListOfPossibleNextWordsOf(String word) {
+        List<String> listOfPossibleNextWords = new ArrayList<String>();
+        for (String considerWord : wordsWithDefinedLetterCount) {
+            if (countNumberOfDifferenceBetweenWords(word, considerWord) == 1) {
+                listOfPossibleNextWords.add(considerWord);
             }
         }
-        return nextWords;
+        return listOfPossibleNextWords;
+
     }
 
-    private int getWordDiffrenceNumber(final String firstWord, String compareWord) {
+    private int countNumberOfDifferenceBetweenWords(final String firstWord, String compareWord) {
```

Comment:
Czy uzywasz tej metody więcej niż raz? Jeśli nie to ja bym ją zrobił bool i zmienił jej nazwę. Łatwiej się wtedy if'y z nazwą tej metody czyta

Reply:
Zmieniłem na boolean